python/ribasim/ribasim/input_base.py

```diff
@@ -30,65 +60,181 @@ def exists(connection: Connection, name: str) -> bool:
-            attr = getattr(self, field)
-            if isinstance(attr, pd.DataFrame):
-                colnames = "(" + ", ".join(attr.columns) + ")"
-                if len(colnames) > 50:
-                    colnames = textwrap.indent(
-                        textwrap.fill(colnames, width=50), prefix="    "
-                    )
-                    entry = f"{field}: DataFrame(rows={len(attr)})\n{colnames}"
-                else:
-                    entry = f"{field}: DataFrame(rows={len(attr)}) {colnames}"
-            else:
-                entry = f"{field}: {attr}"
-            content.append(textwrap.indent(entry, prefix="   "))
-        return "\n".join(content)
+        If a derived FileModel does not load data from disk, this should
+        return an empty dictionary.
 
-    def get_node_IDs(self) -> Set[int]:
-        node_IDs: Set[int] = set()
-        for name in self.fields():
-            attr = getattr(self, name)
-            if isinstance(attr, pd.DataFrame):
-                if "node_id" in attr:
-                    node_IDs.update(attr["node_id"])
+        Args
+        ----
+            filepath (Path): Path to the data to load.
 
-        return node_IDs
+        Returns
+        -------
+            Dict: The data stored at filepath
+        """
+        raise NotImplementedError()
+
+
+class TableModel(FileModel, Generic[TableT]):
+    df: DataFrame[TableT] | None = Field(default=None, exclude=True, repr=False)
+
+    @model_serializer
+    def set_model(self) -> Path | None:
+        return self.filepath
 
     @classmethod
-    def _layername(cls, field) -> str:
-        return f"{cls.get_input_type()}{delimiter}{field}"
+    def tablename(cls) -> str:
+        """Retrieve tablename based on attached Schema.
+
+        NodeSchema -> Schema
+        TabularRatingCurveStaticSchema -> TabularRatingCurve / Static
+        """
+        names: List[str] = re.sub("([A-Z]+)", r" \1", str(cls.tableschema())).split()
+        if len(names) > 2:
+            return f"{''.join(names[:-2])}{delimiter}{names[-2].lower()}"
+        else:
+            return names[0]
+
+    # def __repr__(self) -> str:
+    #     content = [f"<ribasim.{type(self).__name__}>"]
+    #     for field in self.fields():
+    #         attr = getattr(self, field)
+    #         if isinstance(attr, pd.DataFrame):
+    #             colnames = "(" + ", ".join(attr.columns) + ")"
+    #             if len(colnames) > 50:
+    #                 colnames = textwrap.indent(
+    #                     textwrap.fill(colnames, width=50), prefix="    "
+    #                 )
+    #                 entry = f"{field}: DataFrame(rows={len(attr)})\n{colnames}"
+    #             else:
+    #                 entry = f"{field}: DataFrame(rows={len(attr)}) {colnames}"
+    #         else:
+    #             entry = f"{field}: {attr}"
+    #         content.append(textwrap.indent(entry, prefix="   "))
+    #     return "\n".join(content)
```

Comment:
I assume this can be deleted?

Reply:
Yeah. I'm not sure how we want our models to be represented, but it seemed this was some effort, so I commented it out for later adaption.